test/Optimizer/StackArgumentsOptNegativeIndex.js

```diff
@@ -0,0 +1,99 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft Corporation and contributors. All rights reserved.
+// Copyright (c) 2021 ChakraCore Project Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+let fail = 0;
+
+// Test 1 inline negative index
+function func1() {
+    return func2(5);
+}
+
+function func2() {
+    return arguments[-1];
+}
+
+const obj = { "o" : func2 };
+
+for (i = 0; i < 500; i++) {
+    if(func1() != undefined) {
+        ++ fail;
+    }
+}
+
+// Test 2 edited negative index
+function func3() {
+    return func4(5);
+}
+
+function func4() {
+    arguments[-1] = 5;
+    return arguments[-1];
+}
+
+for (i = 0; i < 500; i++) {
```

Comment:
Are the loops added to ensure JITing?

Reply:
To get profile info and then jit - may not be needed if I set it up to work correctly with the dynopogo test setup but this just did the job simply.